src/java.base/share/classes/jdk/internal/misc/InnocuousThread.java

```diff
@@ -104,28 +104,46 @@ public static Thread newSystemThread(String name, Runnable target) {
      */
     public static Thread newSystemThread(String name, Runnable target, int priority) {
         if (System.getSecurityManager() == null) {
-            return createThread(name, target, null, priority);
+            return createThread(name, target, 0L, null, priority);
         }
         return AccessController.doPrivileged(
                 new PrivilegedAction<Thread>() {
                     @Override
                     public Thread run() {
-                        return createThread(name, target, null, priority);
+                        return createThread(name, target, 0L, null, priority);
                     }
                 });
     }
 
-    private static Thread createThread(String name, Runnable target, ClassLoader loader, int priority) {
+    /**
+     * Returns a new InnocuousThread with null context class loader.
+     * Thread priority is set to the given priority.
+     */
+    public static Thread newSystemThread(String name, Runnable target, long stackSize, int priority) {
+        if (System.getSecurityManager() == null) {
+            return createThread(name, target, stackSize, null, priority);
+        }
+        return AccessController.doPrivileged(
+                new PrivilegedAction<Thread>() {
+                    @Override
+                    public Thread run() {
+                        return createThread(name, target, 0L, null, priority);
+                    }
+                });
+    }
+
+    private static Thread createThread(String name, Runnable target, long stackSize, ClassLoader loader, int priority) {
         Thread t = new InnocuousThread(INNOCUOUSTHREADGROUP,
-                target, name, loader);
+                target, name, stackSize, loader);
         if (priority >= 0) {
```

Comment:
Adding an addition newSystemThread method should be okay but can you fix the formatting/line lengths so it's a bit more inconsistent with the existing code.